blog/content/post/2022-12-16-trenchboot-aem-for-qubesos.md

```diff
@@ -0,0 +1,367 @@
+developed to which SINIT ACM will return control. The new entry point was
+responsible for saving information that a TXT launch happened and clean up the
+processor state so that the booting of Xen kernel could continue with the
+standard multiboot path. Among others, if Xen detected TXT launch, it had to
+perform the special processor cores wakeup process (which has been rewritten
+from TrenchBoot Linux patches to Xen native code) and measure external
+components before using them (that is the Xen parameters, Dom0 Linux kernel and
+initrd). Xen also had to reserve the memory regions used by Intel TXT, like it
+was done when tboot has been utilized. The relevant source code for the
+respective Qubes Xen package is available [here](https://github.com/3mdeb/qubes-vmm-xen/pull/1).
+
+# Installation and verification of TrenchBoot AEM on Qubes OS
+
+For a seamless deployment and installation of TrenchBoot AEM, the modifications
+have been converted to patches which are applied to projects' sources during
+Qubes OS components compilation. Those patches have been presented earlier with
+links to Pull Requests. It allows building ready-to-use RPM packages that can
+be installed directly on an installed Qubes OS system. Below a procedure for
+building the packages has been presented. If your are not interested in
+compilation, skip to the [next section](#installing-xen-and-grub-packages).
+The pre-built packages can be downloaded from [here](https://3mdeb.com/open-source-firmware/QubesOS/trenchboot_aem_poc/).
+The packages have been covered with SHA512 sums signed with 3mdeb's
+`Qubes OS TrenchBoot AEM open-source software release 0.x signing key`
+available on [3mdeb-secpack repository](https://github.com/3mdeb/3mdeb-secpack/blob/master/open-source-software/qubes-os-trenchboot-aem-open-source-software-release-0.x-signing-key.asc).
+To verify the RPM packages, fetch the key with:
+`gpg --fetch https://raw.githubusercontent.com/3mdeb/3mdeb-secpack/master/open-source-software/qubes-os-trenchboot-aem-open-source-software-release-0.x-signing-key.asc`
+and then for each package please run:
+
+```txt
+gpg --verify <rpm_package-name>.sha512.sig <rpm_package-name>.sha512
+sha512sum -c <rpm_package-name>.sha512
+```
+
+Check if GPG returns good signature and if yes, check if the RPM checksum
+matches. All files must be in the same directory for the procedure to work.
+
+Note, in order to use the TrenchBoot AEM for Qubes OS you have to own a
+TXT-capable platform with TXT-enabled firmware offering legacy boot. You may
+find such platform and firmware in the [Dasharo with Intel TXT support](https://blog.3mdeb.com/2022/2022-03-17-optiplex-txt/)
+post.
+
+## Building Xen and GRUB packages
+
+To not make the post excessively long the procedure for building packages
+has been put into [TrenchBoot-SDK documentation](https://github.com/TrenchBoot/trenchboot-sdk/blob/3d56ca7b27bb038629fd838819a1050006725a1e/Documentation/build_qubes_packages.md).
```

Comment:
Comment to the content of that guide:
> $ FETCH_CMD="curl --proto '=https' --proto-redir '=https' --tlsv1.2 --http1.1 -sSfL -o" make get-sources

grub-2.06 is now downloaded by default. But if you need to force it anyway, comment on that sdk PR applies.